.changeset/flat-clouds-greet.md

```diff
@@ -0,0 +1,5 @@
+---
+'xstate': major
+---
+
+The `in: ...` property is deprecated for transitions. It is recommended to use `stateIn()` and `stateNotIn()` guard creators instead.
```

Comment:
I wouldnt call this a deprecatiom, but rather a complete replacement of the old API with the new one